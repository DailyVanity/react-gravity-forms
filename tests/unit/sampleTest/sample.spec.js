jest.mock("isomorphic-unfetch");
// jest.mock('node-fetch');

import { render, screen } from "@testing-library/react";
import * as React from "react";
import GravityForm from "../../../src";

import { Response } from "node-fetch";

// use api mocks
import GFGetRequest from "./__mocks__/get.json";

describe("test if a simple newsletter signup works", () => {
  it("should render the form", async () => {
    fetch.mockReturnValue(
      Promise.resolve({ json: () => {JSON.stringify(GFGetRequest)} })
    );
    // dit werkt
    // console.log(await fetch("asd").then(res => res.json()).then(response => response))

    const component = await render(
      <GravityForm
        backendUrl={
          "https://backend.glamrock.dev.gohike.nl/wp/wp-json/glamrock/v1/gf/forms"
        }
        formID={"1"}
        title={"titel"}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
