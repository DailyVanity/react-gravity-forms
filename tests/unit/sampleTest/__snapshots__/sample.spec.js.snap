// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if a simple newsletter signup works should render the form 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-wrapper"
        id="gravity_form_1"
      >
        <form
          novalidate=""
        >
          <div>
            <h3
              class="form-title"
            >
              Test form Mailchimp
            </h3>
            <p
              class="form-description"
            >
              Can we update user
            </p>
          </div>
          <div
            class="form-wrapper"
          >
            <div
              class="form-fields"
            >
              <div
                class="form-field "
              >
                <div
                  class="text"
                >
                  <label
                    class="gf-label "
                    for="input_1_1"
                  >
                    Name
                  </label>
                  <input
                    aria-describedby="error_1_1"
                    aria-invalid="false"
                    aria-label="Name"
                    id="input_1_1"
                    maxlength=""
                    name="input_1"
                    placeholder=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="form-field "
              >
                <div
                  class="email"
                >
                  <label
                    class="gf-label "
                    for="input_1_3"
                  >
                    Mail
                    <abbr>
                      *
                    </abbr>
                  </label>
                  <input
                    aria-describedby="error_1_3"
                    aria-invalid="false"
                    aria-label="Mail"
                    id="input_1_3"
                    maxlength=""
                    name="input_3"
                    placeholder=""
                    required=""
                    type="email"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="footer"
            >
              <input
                name="nonce"
                type="hidden"
                value=""
              />
              <button
                mr="20"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-wrapper"
      id="gravity_form_1"
    >
      <form
        novalidate=""
      >
        <div>
          <h3
            class="form-title"
          >
            Test form Mailchimp
          </h3>
          <p
            class="form-description"
          >
            Can we update user
          </p>
        </div>
        <div
          class="form-wrapper"
        >
          <div
            class="form-fields"
          >
            <div
              class="form-field "
            >
              <div
                class="text"
              >
                <label
                  class="gf-label "
                  for="input_1_1"
                >
                  Name
                </label>
                <input
                  aria-describedby="error_1_1"
                  aria-invalid="false"
                  aria-label="Name"
                  id="input_1_1"
                  maxlength=""
                  name="input_1"
                  placeholder=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-field "
            >
              <div
                class="email"
              >
                <label
                  class="gf-label "
                  for="input_1_3"
                >
                  Mail
                  <abbr>
                    *
                  </abbr>
                </label>
                <input
                  aria-describedby="error_1_3"
                  aria-invalid="false"
                  aria-label="Mail"
                  id="input_1_3"
                  maxlength=""
                  name="input_3"
                  placeholder=""
                  required=""
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="footer"
          >
            <input
              name="nonce"
              type="hidden"
              value=""
            />
            <button
              mr="20"
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
